### ISSUE TITLE
Errors are generated for valid code with function pointers

### ISSUE THREAD

**Issue by Bikeman868** [Dart-Code/Dart-Code#387]
I just found dart-code and VS Code after a long time spent writing Dart without Intellisense. Thank you so much for providing this tool to the community.

My application compiles and works perfectly, and has been in production for a while, but dart-code throws up a few errors that I think are not really errors. This is a stand alone example below:

```
class ConfirmationMessageEvent
{
  String messsage;
  void() confirmAction;
  String confirmButton;
  String cancelButton;

  ConfirmationMessageEvent(
    this.messsage, 
    this.confirmAction, 
    [
      this.confirmButton = 'OK', 
      this.cancelButton = 'Cancel'
    ])
  {}
}
```

**Comment by DanTup** [comment]
I pasted this into [DartPad](https://dartpad.dartlang.org/) and it complaints about this line:

```dart
 void() confirmAction;
```

It seems confused by the `(`. I've never seen this written like this before, so don't know whether it's valid. I had a quick search and can only find examples using `Function`. Changing it to this works:

```dart
 Function confirmAction;
```

So, I'm not sure if this has been changed somewhere along the way, but the Dart analyzer seems to consider it invalid. If you think it should be valid, I think it's worth raising an issue at https://github.com/dart-lang/sdk or maybe posting in the dart-misc group; unfortunately it's not something I can unfluence in Dart Core (since we just call the analyzer from the SDK).
